src/modules/fw_path_navigation/FixedwingPositionControl.cpp

```diff
@@ -1245,7 +1201,7 @@ FixedwingPositionControl::control_auto_loiter(const float control_interval, cons
 
 	} else {
 		// No valid previous waypoint, go for the current wp.
-		// This is automatically handled by the L1/NPFG libraries.
+		// This is automatically handled by the NPFG libraries.
```

Comment:
previous waypoint is not used in this method. would delete this entire if condition.

Reply:
Hmm, this made me think, the navigateLoiter function is not using the previous waypoint. But don't we actually want to fly on the line from the previous waypoint to the center circle, as long as we are "far" away from the circle?

Reply:
> But don't we actually want to fly on the line from the previous waypoint to the center circle, as long as we are "far" away from the circle?

I'd argue not. I don't see any benefit to it. Rather just take the shortest path to the orbit point. But if we want this behavior, I think we should encode it outside NPFG methods. So if we want this as optional.. i dont know if there is param for this or not.. the navigate method could take the line between the prev and current waypoint until reaching `switchDistance()` + `radius` from the orbit center.

Reply:
For sure, this should be outside the npfg. For the moment it would reside inside the auto loiter function. I also did a small test on main, here is the screenshot. 
![Screenshot from 2023-02-07 16-55-03_small](https://user-images.githubusercontent.com/98741601/217296535-5d163d7a-12d8-4fdd-aff8-3395cfa3fb96.png)
So the last loiter is flown to directly, i would have thought it follows the yellow line (in the plan view, it also says it follows the blue line). I would say, if you want to use the shortest path, make the previous invalid, else follow the line, as it is more intuitive as what is shown as mission. What do you guys think?

Reply:
Anyhow, this would be a separate pr and not in here.

Reply:
I think we can get more consistency by always flying directly from the current position to the loiter, instead of from the previous waypoint. Waypoint line following for me only seems important for normal waypoints (eg mapping). @Jaeyoung-Lim what's your take on it?

Reply:
> Waypoint line following for me only seems important for normal waypoints (eg mapping).

I'd also parrot this sentiment. That if a specific line is wished to be flown.. actually plan that specific line with two waypoints. The root issue here is our improper display on QGC, and generally ambiguous behavioral intent. The line from last waypoint in mission to orbit point is misleading. We should rather fix that - and be explicit about what we are intending.

Reply:
Fine by me, even better if the line in GGC vanished, then its also clearer from the mission planning side.